tests/phpunit/tests/test-option-schema.php

```diff
@@ -0,0 +1,481 @@
+			),
+			'post ID string'              => array(
+				'value'           => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => '4',
+					),
+				),
+				'sanitized_value' => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => 4,
+					),
+				),
+				'expected_valid'  => true,
+			),
+			'post ID wrong type'          => array(
+				'value'           => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => array(),
+					),
+				),
+				'sanitized_value' => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => 0,
+					),
+				),
+				'expected_valid'  => new WP_Error( 'rest_invalid_type', '%s[twentyfoobar][fr] is not of type integer.' ),
+			),
+		);
+	}
+
+	public function sync_provider() {
+		return array(
+			'valid'        => array(
+				'value'           => array( 'taxonomies', 'post_meta', 'comment_status' ),
+				'sanitized_value' => array( 'taxonomies', 'post_meta', 'comment_status' ),
+				'expected_valid'  => true,
+			),
+			'valid empty'  => array(
+				'value'           => array(),
+				'sanitized_value' => array(),
+				'expected_valid'  => true,
+			),
+			'valid string' => array(
+				'value'           => 'taxonomies',
+				'sanitized_value' => array( 'taxonomies' ),
+				'expected_valid'  => true,
+			),
+			'wrong type'   => array(
+				'value'           => 8,
+				'sanitized_value' => array( '8' ),
+				'expected_valid'  => new WP_Error( 'rest_not_in_enum', '%s[0] is not one of taxonomies, post_meta, comment_status, ping_status, sticky_posts, post_date, post_format, post_parent, _wp_page_template, menu_order, and _thumbnail_id.' ),
+			),
+			'invalid'      => array(
+				'value'           => array( 'foo' ),
+				'sanitized_value' => array( 'foo' ),
+				'expected_valid'  => new WP_Error( 'rest_not_in_enum', '%s[0] is not one of taxonomies, post_meta, comment_status, ping_status, sticky_posts, post_date, post_format, post_parent, _wp_page_template, menu_order, and _thumbnail_id.' ),
+			),
+		);
+	}
+
+	public function version_provider() {
+		return array(
+			'valid'      => array(
+				'value'           => '1.2.3',
+				'sanitized_value' => '1.2.3',
+				'expected_valid'  => true,
+			),
+			'wrong type' => array(
+				'value'           => 3,
+				'sanitized_value' => '3',
+				'expected_valid'  => new WP_Error( 'rest_invalid_type', '%s is not of type string.' ),
+			),
+		);
+	}
+
+	/**
+	 * @param string        $class           Option class.
+	 * @param mixed         $value           The value to test.
+	 * @param mixed         $sanitized_value Sanitized value.
+	 * @param true|WP_Error $expected_valid  Validation result.
+	 */
+	private function test_option( string $class, $value, $sanitized_value, $expected_valid ) {
+		$option = new $class( $value );
+
+		$this->assertSame( $sanitized_value, $option->get() );
+
+		$valid = rest_validate_value_from_schema( $value, $option->get_schema(), $option->key() );
```

Comment:
Why not using `set()` and `get_errors()` instead?
So the behavior of `PLL_Abstract_Option` is really tested. That way it seems to me we're testing WordPress.
What do you think?

Reply:
This would require to create an instance of `Options`, `PLL_Model`, `PLL_Frontend` (some validations need `PLL()`).
My idea was to focus only on the schemas.

Reply:
`PLL()` and consort could be mocked as well. But as you see fit.